draft-brunstrom-taps-impl.md

```diff
@@ -427,7 +427,7 @@ Protocols that provide the framing (such as length-value protocols, or protocols
 
 ### Sending Messages
 
-The effect of the application sending a Message is determined by the top-level protocol in the established Protocol Stack. That is, if the top-level protocol provides an abstraction of framed messages over a connection, the application will be able to send multiple Messages on that connection, even if the framing protocol is built on a byte-stream protocol like TCP.
+The effect of the application sending a Message is determined by the top-level protocol in the established Protocol Stack. That is, if the top-level protocol provides an abstraction of framed messages over a connection, the receiving application will be able to obtain multiple Messages on that connection without supplying a deframer, even if the framing protocol is built on a byte-stream protocol like TCP.
 
```

Comment:
I don't get it: if you don't supply a deframer for a byte-stream protocol, how does the top-level protocol provide the abstraction of framed messages over the connection? Surely the point of the deframer is to provide this.

Reply:
Surely that's the point of a deframer, but it's also possible without providing one to the transport system: the receiving application can determine message boundaries on its own.

Reply:
If you have HTTP over TCP, then you have an HTTP deframer, surely?

Reply:
So your point is that the receiving application MUST tell the transport system about its deframer?

Reply:
Yes. This doesn't change anything on the wire, or affect interoperability with existing endpoints or protocols, it just means the application gives the de-framing function to the stack to run, rather than doing it itself.

Reply:
I see. Sure, this way, it can work. It wasn't clear to me that, in order to obtain messages, a receiver MUST provide a de-framer. I'm not sure this is very explicit in the text?